opt/scopeserver/dataserver/utils/constant.py

```diff
@@ -273,9 +274,17 @@ class Species(Enum):
 ]
 
 
-def to_colours(index: Iterable[int]) -> List[str]:
+def to_colours(index: Iterable[int], color_list=None) -> List[str]:
     """Convert indexes (`index`) into pre-determined HTML hex colour values.
     `index` can be a `range(some_number)` or any iterable collection of numbers.
     """
-    num_colours = len(BIG_COLOR_LIST)
-    return [BIG_COLOR_LIST[i % num_colours] for i in index]
+    if color_list is not None and len(np.unique(index)) > len(color_list):
+        raise Exception(f"Not enough custom colors defined.")
+
+    COLOR_LIST = (
```

Comment:
I'm not a fan of using the same variable name just different in case, it could be a little confusing.

Maybe `final_color_list` or something similar is better?